frontend/src/pages/modelServing/screens/global/utils.ts

```diff
@@ -15,10 +15,18 @@ export const getInferenceServiceActiveModelState = (
   <InferenceServiceModelState | undefined>is.status?.modelStatus.states.targetModelState ||
   InferenceServiceModelState.UNKNOWN;
 
-export const getInferenceServiceStatusMessage = (is: InferenceServiceKind): string =>
-  is.status?.modelStatus.states.activeModelState ||
-  is.status?.modelStatus.lastFailureInfo?.message ||
-  'Unknown';
+export const getInferenceServiceStatusMessage = (is: InferenceServiceKind): string => {
+  const activeModelState = is.status?.modelStatus.states.activeModelState;
+  const targetModelState = is.status?.modelStatus.states.targetModelState;
+
+  return activeModelState === InferenceServiceModelState.FAILED_TO_LOAD ||
+    targetModelState === InferenceServiceModelState.FAILED_TO_LOAD
+    ? is.status?.modelStatus.lastFailureInfo?.message ??
+        activeModelState ??
+        targetModelState ??
+        'Unknown'
+    : activeModelState ?? targetModelState ?? 'Unknown';
+};
```

Comment:
```suggestion
  const failedToLoad = InferenceServiceModelState.FAILED_TO_LOAD;
  const isFailedToLoad = activeModelState === failedToLoad || targetModelState === failedToLoad;

  const lastFailureMessage = is.status?.modelStatus.lastFailureInfo?.message;
  const stateMessage = activeModelState ?? targetModelState ?? 'Unknown';

  return isFailedToLoad ? lastFailureMessage ?? stateMessage : stateMessage;
```

Try to make the code more legible and avoid redundancy, don't hesitate to use constants to make piece of code managable inside a function, your logic is not wrong, but it takes a lot of time to read and understand what's happening, make the code easier to read.

Reply:
Yes, this refactored code looks more readable. Thanks!